### PR TITLE
[WIP] block.c little clean

### PR DESCRIPTION
**Pull request by sgaragagghu**
Almost finished, just testing

### CODE REVIEW

File: client/block.c [hunk 1]
```diff
@@ -95,11 +93,12 @@ struct orphan_block {
 
 #define get_orphan_list(hash)      (g_orphan_hashtable + ((hash)[0] & (ORPHAN_HASH_SIZE - 1)))
 
+xdag_time_t xdag_era = XDAG_MAIN_ERA;
```
Comment (jonano614) [inline client/block.c]: You removed static modifier. Is it correct?

File: client/block.c [hunk 2]
```diff
@@ -188,26 +156,8 @@ static inline void accept_amount(struct block_internal *bi, xdag_amount_t sum)
...
-
-		while ((ti = bi->ournext) && bi->amount < ti->amount) {
-			bi->ournext = ti->ournext;
-			ti->ourprev = bi->ourprev;
-			bi->ourprev = ti;
-			ti->ournext = bi;
-			*(bi->ournext ? &bi->ournext->ourprev : &ourlast) = bi;
-			*(ti->ourprev ? &ti->ourprev->ournext : &ourfirst) = ti;
-		}
+		order_ourblocks_by_amount(bi);
```
Comment (jonano614) [inline client/block.c]: So, this part is doing something like sorting?

**Comment by jonano614** [comment]
Yes, I was thinking about it too, I don't like xdag functions in utils too. I would suggest something either time.c and amount.c or convert.c in the client folrder.

**Comment by sgaragagghu** [comment]
@xrdavies yes i'll move in time.c and amount.c, i would like to mode these in sync.c since those are related to sync. but i should make function xdag_request_blocks visible from block.c.
What do you think ?

static int request_blocks(xdag_time_t t, xdag_time_t dt)
{
	int i, res = 0;

	if (!g_xdag_sync_on) return -1;

	if (dt <= REQUEST_BLOCKS_MAX_TIME) {
		xdag_time_t t0 = time_limit;

		for (i = 0;
			xdag_info("QueryB: t=%llx dt=%llx", t, dt),
			i < QUERY_RETRIES && (res = xdag_request_blocks(t, t + dt, &t0, add_block_callback)) < 0;
			++i);

		if (res <= 0) {
			return -1;
		}
	} else {
		struct xdag_storage_sum lsums[16], rsums[16];
		if (xdag_load_sums(t, t + dt, lsums) <= 0) {
			return -1;
		}

		xdag_debug("Local : [%s]", xdag_log_array(lsums, 16 * sizeof(struct xdag_storage_sum)));

		for (i = 0;
			xdag_info("QueryS: t=%llx dt=%llx", t, dt),
			i < QUERY_RETRIES && (res = xdag_request_sums(t, t + dt, rsums)) < 0;
			++i);

		if (res <= 0) {
			return -1;
		}

		dt >>= 4;

		xdag_debug("Remote: [%s]", xdag_log_array(rsums, 16 * sizeof(struct xdag_storage_sum)));

		for (i = 0; i < 16; ++i) {
			if (lsums[i].size != rsums[i].size || lsums[i].sum != rsums[i].sum) {
				request_blocks(t + i * dt, dt);
			}
		}
	}

	return 0;
}

// a long procedure of synchronization
static void *sync_thread(void *arg)
{
	xdag_time_t t = 0;

	for (;;) {
		xdag_time_t st = get_timestamp();
		if (st - t >= MAIN_CHAIN_PERIOD) {
			t = st;
			request_blocks(0, 1ll << 48);
		}
		sleep(1);
	}

	return 0;
}

**Pull request closed without merging**